powerscale/helper/synciq_replication_job_helper.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+	powerscale "dell/powerscale-go-client"
+	"fmt"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/constants"
+	"terraform-provider-powerscale/powerscale/models"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// GetSyncIQReplicationJobs gets the list of SyncIQ jobs.
+func GetSyncIQReplicationJobs(ctx context.Context, client *client.Client, filter *models.SyncIQJobFilterModel) (*powerscale.V7SyncJobs, error) {
+	jobParams := client.PscaleOpenAPIClient.SyncApi.ListSyncv7SyncJobs(context.Background())
+	if filter != nil {
+		if !filter.Sort.IsNull() {
+			jobParams = jobParams.Sort(filter.Sort.ValueString())
+		}
+
+		if !filter.Dir.IsNull() {
+			jobParams = jobParams.Dir(filter.Dir.ValueString())
+		}
+
+		if !filter.Limit.IsNull() {
+			jobParams = jobParams.Limit(int32(filter.Limit.ValueInt64()))
+		}
+
+		if !filter.State.IsNull() {
+			jobParams = jobParams.State(filter.State.ValueString())
+		}
+	}
+
+	resp, _, err := jobParams.Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	// Pagination
+	if resp.Resume != nil {
```

Comment:
You also need to check that user has NOT provided `limit`.
Reference logic: https://github.com/dell/terraform-provider-powerscale/commit/e100f6c3be6eb49b1242a5817b8e173f1da218dc#diff-09404796aadc28199b3968e46808a855ecf08170250d9ac48433bc83646af9f8R52